### PR TITLE
RecipeViewFactory and Changes to ViewManager

### PR DESCRIPTION
**Pull request by narenrameshb**
I implemented RecipeViewFactory to initialize both my views and made the necessary changes to ViewManager. Changes must be made to Main.java accordingly.

### CODE REVIEW

File: src/main/java/com/intelliChef/view/ViewManager.java [hunk 1]
```diff
@@ -4,18 +4,21 @@
 import com.intelliChef.entities.Ingredient;
 import com.intelliChef.use_case.IngredientRepository;
 import com.intelliChef.use_case.analyzeImage.AnalyzeImageInteractor;
+import com.intelliChef.entities.Recipe;
```
Comment (mfatir-y) [inline src/main/java/com/intelliChef/view/ViewManager.java:7]: ViewManager shouldn't access entities. Create an interface for this. Look at the code for IngredientListRepository and how it implements an interface for access to ingredients.

File: src/main/java/com/intelliChef/view/RecipeDetailGUI.java [hunk 2]
```diff
@@ -13,7 +13,7 @@
 import java.net.URL;
 
 public class RecipeDetailGUI extends JFrame {
-    private final BackButtonInteractor backButtonInteractor;
+    private BackButtonInteractor backButtonInteractor;
 
     public RecipeDetailGUI(Recipe recipe, RecipeListView recipeListView) {
```
Comment (mfatir-y) [inline src/main/java/com/intelliChef/view/RecipeDetailGUI.java:18]: Views shouldn't access entities. See other note

**Review by mfatir-y — APPROVED** [review]
Rest looks good, just try not to access recipe entity outside your interactors or data access objects.

**Review by EnzaDinza — APPROVED** [review]
Looks great!

**Pull request merged**